src/challenges.js

```diff
@@ -1,45 +1,110 @@
 /* ======================== CallBacks Practice ============================ */
 const each = (elements, cb) => {
-  // Iterates over a list of elements, yielding each in turn to the `cb` function.
-  // This only needs to work with arrays.
+  for (let i = 0; i < elements.length; i++) {
+    cb(elements[i], i);
+  }
 };
 
 const map = (elements, cb) => {
-  // Produces a new array of values by mapping each value in list through a transformation function (iteratee).
-  // Return the new array.
+  const retArr = [];
+  each(elements, (element, i) => {
+    retArr[i] = cb(element);
```

Comment:
Nice way to set things on an array here. Keep track of index, and set the property. But you could also accomplish this with an `array.push()`